code/modules/uplink/uplink_items.dm

```diff
@@ -274,38 +277,38 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 			var/datum/uplink_item/I = uplink_items[category][item]
 			if(src == I || !I.item)
 				continue
-			if(U.telecrystals < I.cost)
+			if(U.red_telecrystals < I.black_cost)
 				continue
 			if(I.limited_stock == 0)
 				continue
 			possible_items += I
 
 	if(possible_items.len)
 		var/datum/uplink_item/I = pick(possible_items)
-		log_uplink("[key_name(user)] purchased a random uplink item from [U.parent]'s uplink with [U.telecrystals] telecrystals remaining")
+		log_uplink("[key_name(user)] purchased a random uplink item from [U.parent]'s uplink with [U.red_telecrystals] telecrystals remaining")
 		SSblackbox.record_feedback("tally", "traitor_random_uplink_items_gotten", 1, initial(I.name))
 		U.MakePurchase(user, I)
 
-/datum/uplink_item/bundles_tc/telecrystal
-	name = "1 Raw Telecrystal"
+/datum/uplink_item/bundles_tc/red_telecrystal
+	name = "1 Raw Red Telecrystal"
 	desc = "A telecrystal in its rawest and purest form; can be utilized on active uplinks to increase their telecrystal count."
-	item = /obj/item/stack/telecrystal
-	cost = 1
+	item = /obj/item/stack/red_telecrystal
+	black_cost = 1
```

Comment:
...wat?

Black telecrystals to get red telecrystals? Are you sure?

Reply:
Er- as a preamble to going through the entire uplink list, i've made everything black_cost

Of course red telecrystals will be purchased with red.